### PR TITLE
feat: add clean resources action [MGT-1833]

### PR DESCRIPTION
**Pull request by WarmVissarutRonaldoDude**
### Description

create composite action that we can reuse anywhere for clean up resources after PR has been deleted/merged.

- delete images from GHCR.
- delete k8s resources immediately instead of wait for 24hrs.

### CODE REVIEW

**Comment by pupudu** [comment]
Looks pretty good to me 🚚  
I will let the @hipagesgroup/platform review the code though

File: clean/action.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+name: clean
```
Comment (viniciusps) [inline clean/action.yaml:1]: I think `clean` would not be too intuitive 🤔 

what about `cleanup-resources` something like that?

File: clean/action.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    default: '2.5.0'
+  argocdAppName:
+    description: ArgoCD app name (ex. opus-pr-test)
+    required: false
+  argocdServer:
+    description: ArgoCD server domain
+    required: false
+  argocdAuthToken:
+    description: ArgoCD auth token
+    required: false
```
Comment (JoshVee) [inline clean/action.yaml:45]: you've got kebab case, snake case _and_ camel case all being used.

standardise on kebab case

**Review by JoshVee — CHANGES_REQUESTED** [review]
see comments

File: clean/action.yaml (hunk 2 above)
Comment (WarmVissarutRonaldoDude) [reply clean/action.yaml]: ah yeah my bad, done.

File: clean/action.yaml (hunk 1 above)
Comment (WarmVissarutRonaldoDude) [reply clean/action.yaml]: yup agreed.

**Comment by WarmVissarutRonaldoDude** [comment]
@viniciusps @JoshVee please help re-review when you guys available, thank you! 🙏

File: cleanup-resources/action.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      with:
+        token: ${{ inputs.gh-token }}
+        organization: hipagesgroup
+        container: ${{ inputs.container }}
+        dry-run: ${{ inputs.dry-run }}
+        older-than: 0
+        untagged: false
+        tag-regex: ^pr\-${{ inputs.pr-number }}\-
+
+    - name: Delete application from ArgoCD
```
Comment (viniciusps) [inline cleanup-resources/action.yaml:59]: Sorry but another comment. 
I am not sure if it is valid but should we add any checks around the app name or branch? like just to make sure no one accidentally passes a prod Argo app name and causes an outage?

Comment (WarmVissarutRonaldoDude) [reply cleanup-resources/action.yaml:59]: hmm should we restrict token to only delete app & resources in `playground` namespace 🤔

Comment (WarmVissarutRonaldoDude) [reply cleanup-resources/action.yaml:59]: and maybe we can do `argocd app get xxx` and check namespace from that app details 🤔 let me see

Comment (WarmVissarutRonaldoDude) [reply cleanup-resources/action.yaml]: @viniciusps done.

Comment (viniciusps) [reply cleanup-resources/action.yaml]: Awesome. Nice Work 🚀

**Review by viniciusps — APPROVED** [review]

**Comment by WarmVissarutRonaldoDude** [comment]
Hi! @JoshVee can you help re-review too, thanks!

**Pull request closed without merging**